driver/sparse_softmax_cross_entropy_with_logits_driver.hpp

```diff
@@ -0,0 +1,498 @@
+{
+    if(!isContiguous)
+        std::swap(inputDim.front(), inputDim.back());
+    std::vector<int> strides(inputDim.size());
+    strides.back() = 1;
+    for(int i = inputDim.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * inputDim[i + 1];
+    if(!isContiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+int SparseSoftmaxCrossEntropyWithLogitsDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw",
+                         'F',
+                         "1",
+                         "Run only Forward SparseSoftmaxCrossEntropyWithLogits (Default=1)",
+                         "int");
+    inflags.AddTensorFlag("input_dim",
+                          'd',
+                          "7x9",
+                          "The dimensional lengths of the input tensors: NxC. Example: 7x9.");
+
+    inflags.AddInputFlag("is-contiguous", 'C', "1", "is-contiguous (Default=1)", "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "1", "Time (Default=1)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int SparseSoftmaxCrossEntropyWithLogitsDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t input_sz  = GetTensorSize(inputDesc);
+    size_t output_sz = GetTensorSize(outputDesc);
+
+    uint32_t ctx = 0;
+
+    input_dev       = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    input_grad_dev  = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    target_dev      = std::make_unique<GPUMem>(ctx, output_sz, sizeof(int));
+    backprop_dev    = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    output_dev      = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+    output_grad_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+
+    input           = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    input_grad      = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    target          = std::vector<int>(output_sz, 0);
+    backprop        = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    output          = std::vector<Tgpu>(output_sz, static_cast<Tgpu>(0));
+    output_grad     = std::vector<Tgpu>(output_sz, static_cast<Tgpu>(0));
+    output_host     = std::vector<Tref>(output_sz, static_cast<Tref>(0));
+    backprop_host   = std::vector<Tref>(input_sz, static_cast<Tref>(0));
+    input_grad_host = std::vector<Tref>(input_sz, static_cast<Tref>(0));
+
+    if(forw == 0 || forw == 1)
+    {
+        for(int i = 0; i < input_sz; i++)
```

Comment:
The `input_sz` is of type `size_t`, but you're using an `int i` for iteration. Could this type mismatch potentially lead to any issues?

Reply:
resolved